### DISCUSSION TITLE
Sources not present for some modules

### DISCUSSION THREAD

**Discussion by sreich** [libktx/ktx#31] [Q&A] [answered]
i'm not familiar enough to know what's causing this, but in intellij for most modules i can browse in my dependencies/ and see the source (incl java doc annotatinons) of most everything..

there are some exceptions, classes that it seems to only have binaries for

- math
-style
-vistyle
-assets(partial)
-actors

the other ones i've tried such as app and log and scene2d are fine. they're all getting pulled in the same way in gradle, so i'm assuming this is some kind of upstream config issue?

**Comment by czyzby** [comment, marked as answer]
This is likely your setup issue. [Check this out](https://stackoverflow.com/questions/12718753/how-to-download-dependency-sources-for-gradle-project-in-idea). All KTX libraries are pushed with the same Gradle task and [it seems that they are present in Maven Central](https://oss.sonatype.org/content/repositories/releases/com/github/czyzby/ktx-math/1.9.6-b1/).

Let me know if you figure this out.

**Comment by sreich** [comment]
hm this is strange indeed. maybe it's just an intellij error (running the prerelease eap). i tried removing mavenLocal, setting the docs/source to download, wiping ~/.gradle/caches and restarting as well as reimporting the project. it just doesn't seem to like those few.

what's even more puzzling is if i go into 'project structure', i can see it pointing to the sources in the local cache.

i'll close this since it's gotta be user error.

**Comment by czyzby** [comment]
Yeah, I don't see how this is a KTX issue, since all source artifacts seem to be uploaded properly. I hope you'll find a solution to your problem.
